apps/wallet/src/ui/app/pages/home/transfer-coin/SendTokenForm.tsx

```diff
@@ -0,0 +1,328 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ArrowRight16 } from '@mysten/icons';
+import { SUI_TYPE_ARG, Coin as CoinAPI } from '@mysten/sui.js';
+import { Field, Form, useFormikContext, Formik } from 'formik';
+import { useMemo, useEffect } from 'react';
+
+import { createValidationSchemaStepOne } from './validation';
+import { Button } from '_app/shared/ButtonUI';
+import BottomMenuLayout, {
+    Content,
+    Menu,
+} from '_app/shared/bottom-menu-layout';
+import { Text } from '_app/shared/text';
+import { IconTooltip } from '_app/shared/tooltip';
+import { AddressInput } from '_components/address-input';
+import Loading from '_components/loading';
+import { parseAmount } from '_helpers';
+import {
+    useCoinDecimals,
+    useFormatCoin,
+    useAppSelector,
+    useIndividualCoinMaxBalance,
+} from '_hooks';
+import {
+    accountAggregateBalancesSelector,
+    accountCoinsSelector,
+} from '_redux/slices/account';
+import { GAS_TYPE_ARG, Coin } from '_redux/slices/sui-objects/Coin';
+import { useGasBudgetInMist } from '_src/ui/app/hooks/useGasBudgetInMist';
+import { InputWithAction } from '_src/ui/app/shared/InputWithAction';
+
+const initialValues = {
+    to: '',
+    amount: '',
+    isPayAllSui: false,
+    // for gas validation purposes
+    // to revalidate when amount changes
+    gasInputBudgetEst: null as number | null,
+};
+
+export type FormValues = typeof initialValues;
+
+export type SubmitProps = {
+    to: string;
+    amount: string;
+    isPayAllSui: boolean;
+    coinIds: string[];
+    gasBudget: number;
+};
+
+export type SendTokenFormProps = {
+    coinType: string;
+    onSubmit: (values: SubmitProps) => void;
+    initialAmount: string;
+    initialTo: string;
+};
+
+export function GasBudgetEstimationComp({
+    coinType,
+    coinDecimals,
+}: {
+    coinType: string;
+    coinDecimals: number;
+}) {
+    const { values, setFieldValue } = useFormikContext<FormValues>();
+
+    const parsedAmount = useMemo(() => {
+        return parseAmount(values?.amount, coinDecimals);
+    }, [values?.amount, coinDecimals]);
+
+    const allCoins = useAppSelector(accountCoinsSelector);
+    const allCoinsOfTransferType = useMemo(
+        () => allCoins.filter((aCoin) => aCoin.type === coinType),
+        [allCoins, coinType]
+    );
+
+    const gasBudgetEstimationUnits = Coin.computeGasBudgetForPay(
+        allCoinsOfTransferType,
+        parsedAmount
+    );
+    const { gasBudget: gasBudgetEstimation, isLoading } = useGasBudgetInMist(
+        gasBudgetEstimationUnits
+    );
+
+    const [formattedGas, gasSymbol] = useFormatCoin(
+        gasBudgetEstimation,
+        GAS_TYPE_ARG
+    );
+
+    // update the gasInputBudgetEst value when the amount changes
```

Comment:
This doesn't sound right. gasBudgetEstimation should change when the amount above changes